lib/AST/Decl.cpp

```diff
@@ -2090,6 +2090,14 @@ bool swift::conflicting(ASTContext &ctx,
   if (!conflicting(sig1, sig2, skipProtocolExtensionCheck))
     return false;
 
+  // Static functions do not conflict with enum element decls if the types
+  // are different
+  if (sig1.IsStaticFunction == sig2.IsEnumElement) {
+    if (sig1Type != sig2Type) {
+      return false;
+    }
+  }
+
   // Functions always conflict with non-functions with the same signature.
   // In practice, this only applies for zero argument functions.
```

Comment:
I think this comment is misleading -- this is the case we're hitting here, and its not a zero argument function.